firewall/firewall_test.go

```diff
@@ -0,0 +1,35 @@
+package firewall
+
+import (
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestCheckIPAccess(t *testing.T) {
+	ip := "192.168.0.1"
+	endpoint := "/api/v1/example"
+
+	// Test the first five accesses for an IP don't exceed the limit
+	for i := 1; i <= 5; i++ {
+		result, _ := IsIpAbusingResources(ip, endpoint+strconv.Itoa(i))
+		if result {
+			t.Errorf("Expected result to be false, got %v for endpoint %s", result, endpoint+strconv.Itoa(i))
```

Comment:
Using `testify/assert` and `testify/require` for these checks will make writing tests easier.